### PR TITLE
Unit test and XML files were added

### PR DESCRIPTION
**Pull request by igorchupin**
Unit tests and XML files were created

### CODE REVIEW

File: src/test/java/shop/CartTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+public class CartTest {
+
+    Cart testCart;
+    RealItem testRealIem;
+    VirtualItem testVirtualItem;
+    private static final int WEIGTHORSIZE = 99;
```
Comment (EugenBorisik) [inline src/test/java/shop/CartTest.java]: WEIGHT_OR_SIZE

File: src/test/java/parser/JsonParserTest.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+    //Values form Data provider
+    @Test (dataProvider = "Unexisting_Pathes",
+           testName = "Parametrized Test: Reading data from unexciting files",
+           groups = {"JsonParser", "Read_From_File", "Critical Path"},
+           priority = 2,
+           expectedExceptions = NoSuchFileException.class)
+
+    void testWithValueSource(String filePathParametrized) {
+        File newFile2 = new File(filePathParametrized);
```
Comment (EugenBorisik) [inline src/test/java/parser/JsonParserTest.java]: Change name

File: src/test/java/parser/JsonParserTest.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    JsonParser testParser;
+    File newFile;
+    Gson gson;
+    int cartNameLength = 10; //Set the Cart name length
+    String cartName;
+    String pathToFolder = "src/main/resources/";
+    String pathToFile;
+
+
+    public String generateCardName () {
```
Comment (EugenBorisik) [inline src/test/java/parser/JsonParserTest.java]: *Cart

Comment (EugenBorisik) [inline src/test/java/parser/JsonParserTest.java]: Make private

File: src/test/java/parser/JsonParserTest.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        for (int i = 0; i <= cartNameLength  ; i++) {
+            int rndChar = 97 + (new Random().nextInt(122 - 97));
+            Character tempChar = (char) rndChar;
+            cartName = cartName + tempChar;
+        }
+        return cartName;
+    }
+
+
+    public Cart addingItemsToCart (Cart cart) {
```
Comment (EugenBorisik) [inline src/test/java/parser/JsonParserTest.java]: Make private

File: src/test/java/parser/JsonParserTest.java [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+
+    //Disabled Test another decision
+    @Test (enabled = false,
+            testName = "Exception because file is read only", groups = {"JsonParser", "Write_to_file", "Critical Path"},
+            priority = 2,
+            expectedExceptions = IOException.class)
+
+    public void writeToFileReadOnly2() throws IOException {
```
Comment (EugenBorisik) [inline src/test/java/parser/JsonParserTest.java]: Rename test method

File: src/test/java/parser/JsonParserTest.java [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+        reader.close();
+
+        Cart actualCart = testParser.readFromFile(testCartFile);
+
+        SoftAssert softAssert = new SoftAssert();
+
+        softAssert.assertEquals (actualCart.getCartName(), expectedCart.getCartName(), "incorrect cart " +
+                        "name value");
+        softAssert.assertEquals (actualCart.getTotalPrice(), expectedCart.getTotalPrice(), "Incorrect total " +
+                        "Price" );
```
Comment (EugenBorisik) [inline src/test/java/parser/JsonParserTest.java]: I forgot to call assertAll, this test will never fail

**Review by EugenBorisik — CHANGES_REQUESTED** [review]
1. Just want to make sure that you understand that usage of priority attribute always bad idea and can be used only when all other options are not suitable.
2. There are a lot of formatting issues in PR. Perform right click on each class and select Reformat Code in context menu
3. In addition, remove unnecessary blank lines in beginning/end of methods + between annotation and method signature. Examples:  JsonParserFromXMLParametresTest, line 36; JsonParserTest, line 103, etc.

File: src/test/java/shop/CartTest.java (hunk 1 above)
Comment (igorchupin) [reply src/test/java/shop/CartTest.java]: The name of the variable was changed

File: src/test/java/parser/JsonParserTest.java (hunk 2 above)
Comment (igorchupin) [reply src/test/java/parser/JsonParserTest.java]: The name of this variable was changed

File: src/test/java/parser/JsonParserTest.java (hunk 3 above)
Comment (igorchupin) [reply src/test/java/parser/JsonParserTest.java]: Name was changed

Comment (igorchupin) [reply src/test/java/parser/JsonParserTest.java]: changed to "private"

File: src/test/java/parser/JsonParserTest.java (hunk 4 above)
Comment (igorchupin) [reply src/test/java/parser/JsonParserTest.java]: changed to "private"

File: src/test/java/parser/JsonParserTest.java (hunk 5 above)
Comment (igorchupin) [reply src/test/java/parser/JsonParserTest.java]: The method was renamed

File: src/test/java/parser/JsonParserTest.java (hunk 6 above)
Comment (igorchupin) [reply src/test/java/parser/JsonParserTest.java]: I forgot =((( Now it was added.

**Comment by igorchupin** [comment]
> 1. Just want to make sure that you understand that usage of priority attribute always bad idea and can be used only when all other options are not suitable.
> 2. There are a lot of formatting issues in PR. Perform right click on each class and select Reformat Code in context menu
> 3. In addition, remove unnecessary blank lines in beginning/end of methods + between annotation and method signature. Examples:  JsonParserFromXMLParametresTest, line 36; JsonParserTest, line 103, etc.

1. I will delete this attribute
2. done
3. done
4 test for deleting Virtual item was added to the "CartTest" class

File: src/test/java/shop/CartTest.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        consoleCaptor = new ConsoleCaptor();
+        testCart.showItems();
+        String resultAfter = consoleCaptor.getStandardOutput().toString();
+        consoleCaptor.close();
+        double totalPriceAfter = testCart.getTotalPrice();
+
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertFalse(resultAfter.contains(testVirtualItem.getName()), "Name of the deleted Virtual " +
+                "item is shown");
+        softAssert.assertTrue(totalPriceBefore < totalPriceAfter, "Total price after deletion is higher or equal");
```
Comment (EugenBorisik) [inline src/test/java/shop/CartTest.java:55]: And what if totalPriceBefore < totalPriceAfter returns true, but totalPrice is still incorrect? Let's verify exact values

**Review by EugenBorisik — APPROVED** [review]

**Pull request merged**